util/runpytest.sh

```diff
@@ -0,0 +1,173 @@
+#
+# The global result is saved in test-XXX.log
+# It will generate one file and one directory:
+#  * unittest-full-$DATE.log : results of unit tests
+#  * integtest-full-$DATE.log: results of the integration tests
+#  * ./integtest-$DATE/ :logs of integration testing (2 files per conf)
+
+# Root path of the repo
+ODEMIS_DIR="$(readlink -m $(dirname $0)/../)"
+ODEMIS_SRC="$ODEMIS_DIR/src/odemis"
+
+# Some basic static analysis
+echo "Total number of lines of code (not including test cases):"
+find "$ODEMIS_SRC" -name "*.py" -a -not -name "*_test.py" -print0 | wc -l --files0-from=- | tail -1
+
+# Not related to tests, but to QA in general: Exceptions usually take only 1 argument
+# So a comma is probably a sign of syntax error and should be replace by a %
+echo "These files might have syntax error when raising an exception:"
+grep -IrE --colour 'raise .+".*%.*",' --include=*.py "$ODEMIS_SRC" "$ODEMIS_DIR"/scripts/ "$ODEMIS_DIR"/plugins/
+echo "---"
+
+echo "These files have old style classes (with \"object\" as parent)"
+grep -IrE --colour "class .+\(\).*:" --include=*.py "$ODEMIS_SRC"
+echo "---"
+
+
+DATE=$(date +%Y%m%d)
+
+# Run all the unit tests that can be found:
+# Every file which is in the pattern /test/*_test.py
+MAXTIME=1800  # 30 min maximum per test case
+
+PYTHONPATH="$ODEMIS_SRC"/../:../Pyro4/src/:"$PYTHONPATH"
+if [ -f /etc/odemis.conf ]; then
+    # use the odemis config if it's available
+    . /etc/odemis.conf
+fi
+export PYTHONPATH
+
+# This environment variable makes the GUI test cases automatically close the test frames
+export NOMANUAL=1
+
+# This environment variable (should) make the driver test not try to use real hardware (only simulator)
+export TEST_NOHW=1
+
+# This environment variable makes the bugreporter test skip test cases that involve ticket creation
+export TEST_NO_SUPPORT_TICKET=1
+
+# A random number, which is always the same, in order to force the dict order in Python 3.
+# This helps in reproducing issues, by making each run a little less different.
+# Note that this doesn't affect the back-end (as it's run in a separate user),
+# and there are many other things that can affect execution order (eg, threads).
+export PYTHONHASHSEED=1567315
+
+mkdir -p ~/development/odemis-testing
+TESTLOG=~/development/odemis-testing/unittest-full-$DATE.log
+TESTSUMMARY=~/development/odemis-testing/pytest-summary-$DATE.log
+# make sure it is full path
+TESTLOG="$(readlink -m "$TESTLOG")"
+TESTSUMMARY="$(readlink -m "$TESTSUMMARY")"
+
+
+# The temporary files are used to save multi line strings (This had issues with bash)
+rm /tmp/filtered_test.txt /tmp/test_summary.txt
+touch /tmp/test_summary.txt /tmp/filtered_test.txt
+
+
+if [ ! -d /var/run/odemisd ] ; then
+    echo  "Need /var/run/odemisd"
+    sudo mkdir -m 777 /var/run/odemisd
+fi
+
+# stop the backend
+sudo odemis-stop
+
+# find the test scripts (should not contain spaces)
+testfiles="$(find "$ODEMIS_SRC" -wholename "*/test/*test.py")"
```

Comment:
Line 30 reads `# Every file which is in the pattern /test/*_test.py`, so shouldn't this pattern be `"*/test/*_test.py"` instead of `"*/test/*test.py"`?

Reply:
I should still catch the same test cases. @pieleric Do you think it is safe to assume all test cases are named correctly and none forgot the "_"?

Reply:
Use `_` everywhere.

Reply:
@tepals you are right (I have discussed it with @pieleric ) I will also fix these comments